pkg/federate/kubefed/resource.go

```diff
@@ -41,16 +46,44 @@ func (f *federator) Distribute(resource runtime.Object, clusterNames ...string)
 	}
 
 	// Update first, as it will probably be the most common operation
-	err = f.kubeFedClient.Update(context.TODO(), fedResource)
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
I think RetryOnConflict should be a consumer decision. Also in our use case there is not going to be need for this and thus its usage is not warranted.

Reply:
I think it's safest to do it this way. It certainly doesn't hurt.

Reply:
just make sure to cover it with tests